src/funcx_common/task_storage/redis.py

```diff
@@ -0,0 +1,60 @@
+import typing as t
+
+from ..tasks import TaskProtocol
+from .base import StorageException, TaskStorage
+
+
+class RedisTaskStorage(TaskStorage):
+    """
+    RedisImplicitTaskStorage stores values in redis by returning the
+    target string in a wrapper so that the storage request is reverse
+    delegated to the caller which stores the task.<fields> into REDIS
+
+    This relies on the caller having a mechanism in place to store
+    values into redis
+    """
+
+    storage_id = "Redis"
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def store_result(self, task: TaskProtocol, result: str) -> bool:
+        task.result = result
+        task.result_reference = {"storage_id": self.storage_id}
+        return True
+
+    def get_result(self, task: TaskProtocol) -> t.Optional[str]:
+        if (
+            task.result_reference
+            and task.result_reference["storage_id"] == self.storage_id
+        ):
```

Comment:
Do we need this check? The point of making these objects chainable is that delegation can happen at another level, and this method can literally just return `task.result`.

I think the resulting `StorageException` below will be unreachable under any valid usages.

Reply:
To clarify, that also applies to the equivalent check in other storage objects.

It may be worth noting that there's no memoization or caching in `RedisField`, so doing an extra fetch of `task.result_reference` may give us a performance hit.

Reply:
It will be reached if there's no result in the current Task model. Current ChainedTaskStorage is managing the logic in this case for now, and I think it is best to keep the backward compatibility changes to one place. The extra fetch is an optimization and we can come around to that.

Reply:
If there's no result in the task, then `task.result` will be `None`. This is testing `task.result_reference`, which is _always_ `None` on existing tasks (because that field does not exist yet). I'll look at the refactor to see if this has changed, but the way this was written, `RedisTaskStorage` would not have been able to read data from existing tasks at all because this check would always fail.